packages/peregrine/lib/talons/WishlistPage/useWishlistItem.js

```diff
@@ -140,13 +173,15 @@ export const useWishlistItem = props => {
     ]);
 
     const handleRemoveProductFromWishlist = useCallback(async () => {
+        setIsDeleting(true);
         try {
             await removeProductsFromWishlist();
 
             // Close the dialogs on success.
             setCurrentDialog(dialogs.NONE);
         } catch (e) {
             console.error(e);
+            setIsDeleting(false);
```

Comment:
Shouldn't this be part of the try block as well? I understand the item will be removed but I think for aesthetics purposes it makes sense to reset it. I might be wrong, maybe it will cause a react update error because the component will be unmounted.

If you think it does cause any error and what I said makes sense, move this to the finally block so it will run irrespective of the item being removed or canceled.

I am fine either way 😄 .

Reply:
finallyBlock dont work strange UI behaivor try differnt things to get nice UI dont work well here

Reply:
So how about having this in the try block as well?

Reply:
### Current behaihor 
![Default Store View](https://user-images.githubusercontent.com/5289370/123785868-4e69ee00-d8d9-11eb-943e-b6e561e6a02f.gif)
### behaihor  after change
![Default Store View](https://user-images.githubusercontent.com/5289370/123785909-5b86dd00-d8d9-11eb-8660-f2d5b5ebf6b7.gif)

Problem to invalidate apollo cache it takes time

Reply:
Cool, then lets stick to what we had before. You can ignore this comment.